README.md

```diff
@@ -1,2 +1,422 @@
 # SafeDI
-Compile-time safe dependency injection in Swift
+
+[![CI Status](https://img.shields.io/github/actions/workflow/status/dfed/SafeDI/ci.yml?branch=main)](https://github.com/dfed/SafeDI/actions?query=workflow%3ACI+branch%3Amain)
+[![Swift Package Manager compatible](https://img.shields.io/badge/SPM-✓-4BC51D.svg?style=flat)](https://github.com/apple/swift-package-manager)
+[![codecov](https://codecov.io/gh/dfed/SafeDI/branch/main/graph/badge.svg)](https://codecov.io/gh/dfed/SafeDI)
+[![License](https://img.shields.io/badge/License-MIT-blue.svg)](https://spdx.org/licenses/MIT.html)
+
+Compile-time safe dependency injection for Swift projects.
+
+## Features
+
+✅ Compile-time safe
+
+✅ Thread safe
+
+✅ Hierarchical dependency scoping
+
+✅ Constructor injection
+
+✅ Multi-module support
+
+✅ Dependency inversion support
+
+✅ Transitive dependency solving
+
+✅ Cycle detection
+
+✅ Architecture independent
+
+✅ Simple integration: no DI-specific types or generics required
+
+✅ Easy testing: every type has a memberwise initializer
+
+✅ Clear error messages: never debug generated code
+
+## Using SafeDI
+
+SafeDI utilizes Swift Macros and Swift Package Manager plugins to read your code and generate a dependency tree that is validated at compile time. Dependencies can either be instantiated by SafeDI or forwarded into the SafeDI dependency tree.
+
+Opting a type into the SafeDI dependency tree is straightforward: add the `@Instantiable` macro to your type declaration, and decorate your type‘s dependencies with macros that signal the lifecycle of each property. Decorate a property with `@Instantiated` if you want to initialize it when the enclosing type is initialized; `@Forwarded` if you want to propagate a runtime-determine value down the dependency tree; or `@Received` if you want to receive the property from an `@Instantiated` or `@Forwarded` property further up the dependency tree.
+
+If a type is declared in third-party code, you can declare an extension the type in your code and decorate it with the `@ExternalInstantiable` macro to opt it into SafeDI.
+
+Let‘s walk through each of these macros in detail.
```

Comment:
The inclusion of graphs enhances clarity regarding the various types. Without prior familiarity with the potency of dependency injection in the tree, comprehending this concept initially might prove challenging.

Reply:
Also, I would suggest to mention the different Macros ahead and then jump into an example of each.

Reply:
The intro is fine in this part

Reply:
See the new cheat sheet section, and the updated "Using SafeDI" section. I think I've addressed what you're getting at. Let me know!

Reply:
alright undid the cheat sheet sectioned put in a table at the top.